ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/rtb/VungleRtbBannerAd.java

```
@@ -0,0 +1,121 @@
+      return;
+    }
+
+
+    String placementForPlay = VungleManager
+        .getInstance().findPlacement(mediationExtras, serverParameters);
+    Log.d(TAG,
+        "requestBannerAd for Placement: " + placementForPlay + " ### Adapter instance: " + this
+            .hashCode());
+
+    if (TextUtils.isEmpty(placementForPlay)) {
+      AdError error = new AdError(ERROR_INVALID_SERVER_PARAMETERS,
+          "Failed to load ad from Vungle. Missing or Invalid placement ID.", ERROR_DOMAIN);
+      Log.w(TAG, error.getMessage());
+      mediationAdLoadCallback.onFailure(error);
+      return;
+    }
+
+    Context context = mediationBannerAdConfiguration.getContext();
+    AdSize adSize = mediationBannerAdConfiguration.getAdSize();
+
+    AdConfig adConfig = VungleExtrasBuilder.adConfigWithNetworkExtras(mediationExtras, true);
+    if (!VungleManager.getInstance().hasBannerSizeAd(context, adSize, adConfig)) {
+      AdError error = new AdError(ERROR_BANNER_SIZE_MISMATCH,
+          "Failed to load ad from Vungle. Invalid banner size.", ERROR_DOMAIN);
+      Log.w(TAG, error.getMessage());
+      mediationAdLoadCallback.onFailure(error);
+      return;
+    }
+
+    AdapterParametersParser.Config config = AdapterParametersParser.parse(appID, mediationExtras);
+    // Adapter does not support multiple Banner instances playing for same placement except for
+    // refresh.
+    String uniqueRequestId = config.getRequestUniqueId();
+    if (!VungleManager.getInstance().canRequestBannerAd(placementForPlay, uniqueRequestId)) {
+      AdError error = new AdError(ERROR_AD_ALREADY_LOADED,
+          "Vungle adapter does not support multiple banner instances for same placement.",
+          ERROR_DOMAIN);
+      Log.w(TAG, error.getMessage());
+      mediationAdLoadCallback.onFailure(error);
+      return;
+    }
+
+    String adMarkup = mediationBannerAdConfiguration.getBidResponse();
+    if (TextUtils.isEmpty(adMarkup)) {
+      adMarkup = null;
+    }
```

==== BLOCKING comment =====

If the `adMarkup` is empty, that means it's not a bidding banner ad. Shouldn't we return a failure callback instead?

---

Updated, just like other ad formats removed this check. We never get an empty adMarkup if running into here.